contracts/src/libraries/PolynomialEval.sol

```diff
@@ -14,7 +14,8 @@ library PolynomialEval {
     struct EvalDomain {
         uint256 logSize; // log_2(self.size)
         uint256 sizeInv; // Inverse of the size in the field
-        uint256 groupGen; // A generator of the subgroup
+        uint256 groupGen; // A generator g of the subgroup
+        uint256[8] domainElements; // 1, g, g^2, ..., g^7
```

Comment:
- we can remove `groupGen`
- we can reduce to only 7 and starts with `g` since 1 is the same, we don't have to store it everywhere.

Reply:
> we can reduce to only 7 and starts with g since 1 is the same, we don't have to store it everywhere.

this is bad advice, because it complicates the code, affect readability. an extra uint256 in memory is no big deal.

Reply:
done in [5708d75](https://github.com/EspressoSystems/espresso-sequencer/pull/1860/commits/5708d754124d98b3f1472ae5919633cb8d7419a0)